### ISSUE TITLE
Adding spinner or searching

### ISSUE THREAD

**Issue by bardam** [angular-ui/ui-select#350]
Is there on option to display a spinner or "Searching..." text till the refresh method finished?
I'm loading the data remotely and it can take 2-3 seconds till I get the data.

**Comment by MBentley1976** [comment]
I need this functionality as well.  I need it to go one step further as well.  I'm using a remote data source to ajax in search results.  I need a way to display what amounts to a "no records found" message when the search returns no results.

**Comment by amcdnl** [comment]
@MBentley1976 can you provide an example?  Thinking this should be included in base.

**Comment by seinavco** [comment]
Any updates?

**Comment by codetrash** [comment]
Heres my solution....

```
/**
	 * Get Data products...will be Injected into products
	 */
	$scope.products = [];
	$scope.isProductFound  = false;
	$scope.getProducts = function(search) {
		
		if(!search) return;
		
		$scope.isProductLoaded = true; //for loading info when a search for product is conducted		
		var params = {search: search, warehouse_id : $scope.warehouse.selectedWarehouseId};
		
	    return $http.get(SITE_URL + 'transaction/getProducts', {params: params} ).then(function(response) {
	    
	    	$scope.products = response.data.data
	    	if (response.data.data.length < 1) {
	    		
	    		$scope.isProductFound  = true;
	    	} else {
	    		
	    		$scope.isProductFound  = false;
	    	}
	    	
	    	$scope.isProductLoaded = false;
	    });
    };
```

```
<div class="input-group">
 <ui-select name="product" ng-model="product.no_int_product" on-select=onSelectProduct($item) theme="bootstrap" reset-search-input="true" style="width:500px">
    <ui-select-match placeholder="Search product by Name, Code..">{{$select.selected.product_name}}</ui-select-match>
    <ui-select-choices repeat="product.no_int_product as product in products"
                   refresh="getProducts($select.search)"
                    refresh-delay="500" style="width:500px">
      <div ng-bind-html="product.product_name"></div>
      </ui-select-choices>
</ui-select>
<!-- Show only when loading... -->
<span class="input-group-addon" ng-show="isProductLoaded && !isProductFound" style="background: none;border:none"><i class="fa fa-spinner fa-pulse"></i></span>
<!-- Show only If product not found -->
<span class="input-group-addon" style="background: none;border:none" ng-show="isProductFound">Not found...</span>
</div>
```

**Comment by simonv3** [comment]
FWIW I think this is now here: https://github.com/angular-ui/ui-select/pull/1779

**Comment by user378230** [comment]
Closing as resolved by #1779
